easysid.cpp

```diff
@@ -26,7 +41,15 @@ EXPORT void start_capture(const char *fn)
         delete wav;
         wav = NULL;
     }
-    wav = makePcmOutputWav(fn);
+    if (pcm == NULL) {
+        if (not initAudioOutput(DEFAULT_SAMPLING_RATE)) {
+            fprintf(stderr, "Failed to initialize audio output device: %d\n");
+            perror(NULL);
```

Comment:
Since `perror()` relies on the value of `errno`, you can only call it immediately after the function that actually returned an error code.